docs/RFCS/select_from_index.md

```diff
@@ -0,0 +1,55 @@
+- Feature Name: select_from_index
+- Status: draft
+- Start Date: 2015-08-10
+- RFC PR:
+- Cockroach Issue: #2046
+
+# Summary
+
+Extend the SELECT syntax to allow an optional index to be specified in
+table references. The current table reference syntax is `table` or
+`database.table`. This syntax would be extended to allow `table.index`
+and `database.table.index`.
+
+# Motivation
+
+As indexes are optional in SQL, it is difficult to know when an index
+is used to satisfy a query which in turn makes it difficult to use the
+existing logic test infrastructure to test correctness of index
+scans. Providing for specification of an index to read from will make
```

Comment:
No, that would be invalid. Only the columns that are part of the index are available to the rest of the query. I completely agree that we'll want more control than this proposal provides, but there are still huge swaths of query planning that we're not handling. We currently can't read from an index at all. Once we read from an index I want to be able to handle queries that are satisfied solely by the index. Then queries that use the index and lookup the full row by primary key.